components/identity-core/org.wso2.carbon.identity.core/src/test/java/org/wso2/carbon/identity/core/util/IdentityConfigParserTest.java

```diff
@@ -0,0 +1,102 @@
+
+public class IdentityConfigParserTest {
+
+    @AfterClass
+    public void tearDown() throws Exception {
+        System.clearProperty(ServerConstants.CARBON_HOME);
+    }
+
+    @Test
+    public void testGetInstance() throws Exception {
+        String identityXmlPath = Paths.get(System.getProperty("user.dir"), "src", "test", "resources",
+                "identity.xml").toString();
+        System.setProperty(ServerConstants.CARBON_HOME, ".");
+        IdentityConfigParser.getInstance(identityXmlPath);
+        assertTrue(MapUtils.isNotEmpty(IdentityConfigParser.getInstance().getConfiguration()),"Configuration should " +
+                "not be null/empty after initialization");
+    }
+
+    @Test(dependsOnMethods = "testGetInstance")
+    public void testGetConfiguration() throws Exception {
+        assertTrue(MapUtils.isNotEmpty(IdentityConfigParser.getInstance().getConfiguration()),"Configuration should " +
+                "not be null/empty");
+    }
+
+    @Test(dependsOnMethods = "testGetInstance")
+    public void testGetEventListenerConfiguration() throws Exception {
+        assertTrue(MapUtils.isNotEmpty(IdentityConfigParser.getEventListenerConfiguration()),"Event listener config " +
+                "should not be null/empty");
+    }
+
+    @Test(dependsOnMethods = "testGetInstance")
+    public void testGetIdentityCacheConfigurationHolder() throws Exception {
+        assertTrue(MapUtils.isNotEmpty(IdentityConfigParser.getIdentityCacheConfigurationHolder()),"Cache config " +
+                "holder should not be null/empty");
+    }
+
+    @Test(dependsOnMethods = "testGetInstance")
+    public void testGetIdentityCookieConfigurationHolder() throws Exception {
+        assertTrue(MapUtils.isNotEmpty(IdentityConfigParser.getIdentityCookieConfigurationHolder()),"Cookie config " +
+                "holder should not be null/empty");
+    }
+
+    @DataProvider(name = "ConfigElementValues")
+    public Object[][] getConfigElement() {
+        return new Object[][]{
+                {"ClockSkew", "300"},
+                {"EnableAskPasswordAdminUI", "false"},
+        };
+    }
+
+    @Test(dependsOnMethods = "testGetInstance", dataProvider = "ConfigElementValues")
+    public void testGetConfigElement(String localPart, String expected) throws Exception {
+        assertEquals(IdentityConfigParser.getInstance().getConfigElement(localPart).getText(), expected,"Config " +
+                "provided doesn't match the expected.");
+    }
+
+    @Test(dependsOnMethods = "testGetInstance")
+    public void testGetConfigElementNegative() throws Exception {
+        assertNull(IdentityConfigParser.getInstance().getConfigElement("NoneExisting"),"Should return null for none" +
+                " existing elements");
+    }
+
+    @Test(dependsOnMethods = "testGetInstance")
+    public void testGetQNameWithIdentityNS() throws Exception {
+        QName element = IdentityConfigParser.getInstance().getQNameWithIdentityNS("TestElement");
```

Comment:
Shouldn't the `TestElement` be in the `identity.xml` ?

Reply:
No